frontend/packages/kubevirt-plugin/integration-tests/tests/models/baseVMBuilder.ts

```diff
@@ -0,0 +1,145 @@
+import * as _ from 'lodash';
+import { OperatingSystem, Workload, OSIDLookup } from '../utils/constants/wizard';
+import { FlavorConfig, ProvisionSource, Disk, Network, CloudInitConfig } from '../types/types';
+import { BaseVMBuilderData } from '../types/vm';
+import { K8sKind } from '@console/internal/module/k8s';
+import { getRandStr } from '../utils/utils';
+
+export abstract class BaseVMBuilder<T extends BaseVMBuilderData> {
+  protected kind: K8sKind;
+
+  protected data: T;
+
+  constructor(kind: K8sKind, builder?: BaseVMBuilder<T>) {
+    this.kind = kind;
+    this.data = builder
+      ? builder.getData()
+      : ({ networks: [], disks: [], storageClassAttributes: {} } as any);
+  }
+
+  public setName(name: string) {
+    this.data.name = name;
+    return this;
+  }
+
+  public generateName(id?: string) {
+    const finalId = id || getRandStr(5);
+    const osId = OSIDLookup[this.data.os];
+    this.data.name = [
```

Comment:
Yes,

Reply:
ok, but we should still allow for simple `setName`

Reply:
Good point... What if we add `setName` method and generate random name by default if not set explicitly

Reply:
can we still create some variable like generateNameID and use it when generating name in build method when name is not filled? 
and not run generating name logic here (just from build method) as not all info for it might not be available yet

Reply:
I added 
```
generateNameForPrefix(prefix: string) {
  this.data.name = `${prefix}-${getRandStr(5)}`;
  return this;
{
```
And don't call generateName anywhere except the build methods

Reply:
looks good